app/assets/javascripts/dashboard.js

```diff
@@ -112,7 +112,7 @@ MinionPoller = {
         handleBootstrapButtonTitle();
 
         // show/hide "update all nodes" link
-        $("#update-all-nodes").toggleClass('hidden', !(updateAvailable && allApplied));
+        $("#update-all-nodes").toggleClass('hidden', !updateAvailable || !allApplied);
```

Comment:
I didn't know when update failed but retryable the highstate changes to `failed` as your json shows. So it has to be updated for that case. Will update the spec and code.